internal/resources/slo/resource_slo.go

```diff
@@ -0,0 +1,596 @@
+package slo
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+	"github.com/grafana/terraform-provider-grafana/internal/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceSlo() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+Resource manages Grafana SLOs. 
+
+* [Official documentation](https://grafana.com/docs/grafana-cloud/slo/)
+* [API documentation](https://grafana.com/docs/grafana-cloud/slo/api/)
+		`,
+		CreateContext: resourceSloCreate,
+		ReadContext:   resourceSloRead,
+		UpdateContext: resourceSloUpdate,
+		DeleteContext: resourceSloDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Name should be a short description of your indicator. Consider names like "API Availability"`,
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Description is a free-text field that can provide more context to an SLO.`,
+			},
+			"query": &schema.Schema{
```

Comment:
Because of discussions we've had in the team about the various query schemas, let's strip out everything *except* freeform from this schema. That will protect the provider from any near-term refactoring that we have to do.